src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs

```diff
@@ -0,0 +1,413 @@
+
+                return new DynamicMetaObject(expression, restrictions);
+            }
+            catch (RuntimeBinderException e)
+            {
+                if (onBindingError != null)
+                {
+                    return onBindingError;
+                }
+
+                return new DynamicMetaObject(
+                    Expression.Throw(
+                        Expression.New(
+                            typeof(RuntimeBinderException).GetConstructor(new Type[] { typeof(string) }),
+                            Expression.Constant(e.Message)
+                        ),
+                        GetTypeForErrorMetaObject(action, args.FirstOrDefault())
+                    ),
+                    restrictions
+                );
+            }
+        }
+
+        /////////////////////////////////////////////////////////////////////////////////
+
+        private static bool IsTypeOfStaticCall(
+            int parameterIndex,
+            ICSharpInvokeOrInvokeMemberBinder callPayload)
+        {
+            return parameterIndex == 0 && callPayload != null && callPayload.StaticCall;
+        }
+
+        /////////////////////////////////////////////////////////////////////////////////
+
+        private static bool IsComObject(object obj)
+        {
+            return obj != null && Marshal.IsComObject(obj);
+        }
+
+        /////////////////////////////////////////////////////////////////////////////////
+
+        // Try to determine if this object represents a WindowsRuntime object - i.e. it either
+        // is coming from a WinMD file or is derived from a class coming from a WinMD.
+        // The logic here matches the CLR's logic of finding a WinRT object.
+        internal static bool IsWindowsRuntimeObject(DynamicMetaObject obj)
+        {
+            if (obj != null && obj.RuntimeType != null)
+            {
+                Type curType = obj.RuntimeType;
+                while (curType != null)
+                {
+                    if (curType.GetTypeInfo().Attributes.HasFlag(System.Reflection.TypeAttributes.WindowsRuntime))
+                    {
+                        // Found a WinRT COM object
+                        return true;
+                    }
+                    if (curType.GetTypeInfo().Attributes.HasFlag(System.Reflection.TypeAttributes.Import))
+                    {
+                        // Found a class that is actually imported from COM but not WinRT
+                        // this is definitely a non-WinRT COM object
+                        return false;
+                    }
+                    curType = curType.GetTypeInfo().BaseType;
+                }
+            }
+            return false;
+        }
+
+        /////////////////////////////////////////////////////////////////////////////////
+
+        private static bool IsTransparentProxy(object obj)
+        {
+            return obj != null; // && RemotingServices.IsTransparentProxy(obj);
```

Comment:
Feel free to do this separately, but you can probably just remove the comment entirely. Seems to be related to Remoting, which isn't supported in .NET Core, and we've just been removing such things from other parts of the source.